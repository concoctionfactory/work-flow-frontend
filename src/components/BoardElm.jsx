import React, { useState } from "react";
import { Link } from "react-router-dom";
import { useDispatch } from "react-redux";
import { removeBoardAPI } from "../actions/boards";
import BoardForm from "./BoardForm";

import { makeStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardHeader from "@material-ui/core/CardHeader";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import MoreVertIcon from "@material-ui/icons/MoreVert";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
const useStyles = makeStyles({
  card: {
    height: "100%",
    display: "flex",
    flexDirection: "column",
  },
});

function BoardElm({ board, username }) {
  const classes = useStyles();
  const [anchorEl, setAnchorEl] = React.useState(null);
  const dispatch = useDispatch();

  const handleClick = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  function handleDelete() {
    dispatch(removeBoardAPI(board.id));
    handleClose();
  }

  const [boardForm, setBoardForm] = useState({
    isOpen: false,
  });

  const openBoardFrom = () => {
    handleClose();
    setBoardForm({ isOpen: true });
  };

  const closeBoardFrom = () => {
    setBoardForm({ isOpen: false });
  };

  return (
    <Card className={classes.card}>
      <CardHeader
        action={
          <IconButton
            aria-label="settings"
            aria-haspopup="true"
            onClick={handleClick}
          >
            <MoreVertIcon />
          </IconButton>
        }
      />

      <Menu
        id="simple-menu"
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleClose}
      >
        <MenuItem onClick={openBoardFrom}>Edit</MenuItem>
        <MenuItem onClick={handleDelete}>Delete</MenuItem>
      </Menu>

      <CardActionArea
        className={classes.root}
        component={Link}
        to={`boards/${board.id}`}
      >
        <CardContent>
          <Typography>{board.name}</Typography>
        </CardContent>
      </CardActionArea>
      <BoardForm
        className={classes.fab}
        isOpen={boardForm.isOpen}
        handleOpen={openBoardFrom}
        handleClose={closeBoardFrom}
        username={username}
        board={board}
      ></BoardForm>
    </Card>
  );
}

export default BoardElm;
