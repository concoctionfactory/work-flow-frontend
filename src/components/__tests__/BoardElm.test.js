import React from "react";
import { render } from "@testing-library/react";
import BoardElm from "../BoardElm";
import { userData, boardData } from "./testData";
import { Provider } from "react-redux";
import { createStore, applyMiddleware } from "redux";
import rootReducer from "../../reducers/root";
import { BrowserRouter } from "react-router-dom";
import thunk from "redux-thunk";
const store = createStore(rootReducer, applyMiddleware(thunk));
const username = userData.username;

it("renders without crashing", function () {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <BoardElm board={boardData} username={username} />
      </BrowserRouter>
    </Provider>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <BoardElm board={boardData} username={username} />
      </BrowserRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("shows text ", function () {
  const component = render(
    <Provider store={store}>
      <BrowserRouter>
        <BoardElm board={boardData} username={username} />
      </BrowserRouter>
    </Provider>
  );
  expect(component.getByText("Edit")).toBeInTheDocument();
  expect(component.getByText("Delete")).toBeInTheDocument();
});
