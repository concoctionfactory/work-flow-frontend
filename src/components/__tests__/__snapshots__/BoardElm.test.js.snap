// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-card-2 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content"
      />
      <div
        class="MuiCardHeader-action"
      >
        <button
          aria-haspopup="true"
          aria-label="settings"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiCardActionArea-root"
      href="/boards/91"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          ant site
        </p>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </a>
    <div />
  </div>
</DocumentFragment>
`;
